src/rgw/rgw_zone.h

```diff
@@ -60,7 +62,7 @@ struct RGWZoneParams {
   const std::string& get_compression_type(const rgw_placement_rule& placement_rule) const;
   
   void encode(bufferlist& bl) const {
-    ENCODE_START(16, 1, bl);
+    ENCODE_START(17, 1, bl);
```

Comment:
@cbodley @adamemerson .. while trying to backport these changes to [tentacle](https://github.com/ceph/ceph/blob/tentacle/src/rgw/driver/rados/rgw_zone.h#L156), I observed that the ENCODE/DECODE version will be 16 for restore_pool (same as in downstream 8.1 codebase) in that branch as opposed to 17 in main . Would this cause any issue during upgrade later from tentacle to next upcoming release?

To resolve it, can we include "restore_pool" along with "dedup_pool" in the same ENCODE version i.e, 16 here in this PR?

Reply:
thanks @soumyakoduri 

upstream squid is on the v15 encoding and has up to group_pool: https://github.com/ceph/ceph/blob/squid/src/rgw/driver/rados/rgw_zone.h#L156-L184

i see that https://github.com/ceph/ceph/pull/64264 is pending backport to tentacle, adding v16 for `dedup_pool` (cc @benhanokh)

does 8.1 add `dedup_pool` for v16 too? if not, we'd probably need tentacle to include a v16 with only `restore_pool` to match 8.1, and bump the `dedup_pool` change to v17

Reply:
https://github.com/ceph/ceph/pull/64360 is the backport of this PR to tentacle and [PR64361](https://github.com/ceph/ceph/pull/64361) fixes the versions in `main`